src/engine/GroupBy.cpp

```diff
@@ -801,76 +794,71 @@ bool GroupBy::computeGroupByForSortedSubtree(IdTable* result,
   evaluationContext._isPartOfGroupBy = true;
 
   size_t blockSize = 65536;
-  auto numBlocks = (size_t)ceil(((double)subresult.size()) / blockSize);
 
-  size_t rowIndex = 0;
-
-  for (size_t i = 0; i < numBlocks; ++i) {
-    evaluationContext._beginIndex = i * blockSize;
-    evaluationContext._endIndex =
-        std::min((i + 1) * blockSize, subresult.size());
+  for (size_t i = 0; i < subresult.size(); i += blockSize) {
+    evaluationContext._beginIndex = i;
+    evaluationContext._endIndex = std::min(i + blockSize, subresult.size());
 
     auto currentBlockSize =
         evaluationContext._endIndex - evaluationContext._beginIndex;
 
-    size_t aggregateIndex = 0;
-    auto currentRowIndex = rowIndex;
-
     // Perform HashMap lookup for all groups in current block
-    std::vector<std::array<AverageAggregationData, numAggregates>*> hashEntrys;
+    std::vector<std::array<AverageAggregationData, numAggregates>*> hashEntries;
+
+    auto getId = [&subresult, &evaluationContext, &columnIndex](size_t j) {
+      return subresult(evaluationContext._beginIndex + j, columnIndex);
+    };
 
     // Create elements in map
     for (size_t j = 0; j < currentBlockSize; ++j) {
-      auto id = subresult(evaluationContext._beginIndex + j, columnIndex);
+      auto id = getId(j);
       (void)map[id];
```

Comment:
I think `map.try_emplace(id)` is a little bit more idiomatic (looks more like inserting into a map).